### PR TITLE
Fix `set_module_tensor_to_device`

### PR DESCRIPTION
**Pull request by sgugger**
It turns out that `to` converts a parameter to a tensor, so the else branch that was added in #539 breaks a lot of things 😬 
This PR fixes that.

cc @younesbelkada

### CODE REVIEW

**Review by muellerzr — APPROVED** [review]

File: src/accelerate/utils/modeling.py [hunk 1]
```diff
@@ -127,13 +127,11 @@ def set_module_tensor_to_device(
 
         if is_buffer:
             module._buffers[tensor_name] = new_value
-        elif module._parameters[tensor_name].device.type != "cuda":
+        elif device != module._parameters[tensor_name].device:
```
Comment (younesbelkada) [inline src/accelerate/utils/modeling.py:130]:
```suggestion
        elif device != module._parameters[tensor_name].device.index:
```
The `device` param is an integer so to properly compare the devices we need to access the index of the `device` attribute of the parameter ;)

Comment (sgugger) [reply src/accelerate/utils/modeling.py:130]: Ah should be `torch.device(device)` on the left side to cover all cases!

Comment (sgugger) [reply src/accelerate/utils/modeling.py]: Fixed [here](https://github.com/huggingface/accelerate/pull/576/commits/10334600357bdd014acd67bb255884ee10b54fe6) normally.

Comment (younesbelkada) [reply src/accelerate/utils/modeling.py]: Can confirm this fixes the issue too !

**Pull request merged**